python/lsst/ts/ess/common/abstract_command_handler.py

```diff
@@ -0,0 +1,215 @@
+from .constants import Command, Key, ResponseCode
+from .schema import CONFIG_JSCHEMA
+
+
+class AbstractCommandHandler(ABC):
+    """Handle incoming commands and send replies. Apply configuration and read
+    sensor data.
+
+    Parameters
+    ----------
+    callback: `Callable`
+        The callback coroutine handling the sensor telemetry. This can be a
+        coroutine that sends the data via a socket connection or a coroutine in
+        a test class to verify that the command has been handled correctly.
+    simulation_mode: `int`
+        Indicating if a simulation mode (> 0) or not (0) is active.
+
+    The commands that can be handled are:
+
+        configure: Load the configuration that is passed on with the command
+        and connect to the devices specified in that configuration. This
+        command can be sent multiple times before a start is received and only
+        the last configuration is kept.
+        start: Start reading the sensor data of the connected devices and send
+        it as plain text via the socket. If no configuration was sent then the
+        start command is ignored. Once started no configuration changes can be
+        done anymore.
+        stop: Stop sending sensor data and disconnect from all devices. Once
+        stopped, configuration changes can be done again and/or reading of
+        sensor data can be started again.
+
+    """
+
+    valid_simulation_modes = (0, 1)
+
+    @abstractmethod
+    def __init__(self, callback: typing.Callable, simulation_mode: int) -> None:
+        self.log = logging.getLogger(type(self).__name__)
+        if simulation_mode not in self.valid_simulation_modes:
+            raise ValueError(
+                f"simulation_mode={simulation_mode} "
+                f"not in valid_simulation_modes={self.valid_simulation_modes}"
+            )
+
+        self.simulation_mode = simulation_mode
+
+        self._callback = callback
+        self._configuration: typing.Optional[typing.Dict[str, typing.Any]] = None
+        self._started = False
+
+        self.dispatch_dict: typing.Dict[str, typing.Callable] = {
+            Command.CONFIGURE: self.configure,
+            Command.START: self.start_sending_telemetry,
+            Command.STOP: self.stop_sending_telemetry,
+        }
+
+    async def handle_command(self, command: str, **kwargs: typing.Any) -> None:
+        """Handle incomming commands and parameters.
+
+        Parameters
+        ----------
+        command: `str`
+            The command to handle.
+        kwargs:
+            The parameters to the command.
+        """
+        self.log.info(f"Handling command {command} with kwargs {kwargs}")
+        func = self.dispatch_dict[command]
+        try:
+            await func(**kwargs)
+            response = {Key.RESPONSE: ResponseCode.OK}
+        except CommandError as e:
+            self.log.exception("Encountered a CommandError.")
+            response = {Key.RESPONSE: e.response_code}
```

Comment:
Yes, that is a good suggestion. Raising the exception is good because if something goes wrong then this indicates that starting a device failed which shouldn't happen. Done.